src/analyses/invariant_propagation.cpp

```diff
@@ -15,6 +15,39 @@ Author: Daniel Kroening, kroening@kroening.com
 #include <util/symbol_table.h>
 #include <util/std_expr.h>
 
+/// Pass the necessary arguments to the invariant_set_domaint's when constructed
+class invariant_set_domain_factoryt
+  : public ai_domain_factoryt<invariant_set_domaint>
+{
+public:
+  explicit invariant_set_domain_factoryt(invariant_propagationt &_ip) : ip(_ip)
+  {
+  }
+
+  std::unique_ptr<statet> make_domain(locationt l) const override
+  {
+    auto p = util_make_unique<invariant_set_domaint>(
+      ip.value_sets, ip.object_store, ip.ns);
+    CHECK_RETURN(p->is_bottom());
+
+    return p;
+  }
+
+private:
+  invariant_propagationt &ip;
```

Comment:
Why is this a reference rather than a constant pointer like in the other factory?

Reply:
Ideally they should all be references but I wanted to keep the changes as minimal and obviously correct as possible.